src/osx/carbon/cursor.cpp

```diff
@@ -278,7 +279,26 @@ wxCursor::wxCursor(const wxString& cursor_file, wxBitmapType flags, int hotSpotX
     if ( flags == wxBITMAP_TYPE_MACCURSOR_RESOURCE )
     {
 #if wxOSX_USE_COCOA
-        wxFAIL_MSG( wxT("Not implemented") );
+        wxImage image;
+        wxFileName fileName( wxStandardPaths::Get().GetResourcesDir(), cursor_file, "png" );
+        if( image.LoadFile( fileName.GetFullPath(), wxBITMAP_TYPE_PNG ) )
+        {
+            image.SetOption( wxIMAGE_OPTION_CUR_HOTSPOT_X, hotSpotX ) ;
+            image.SetOption( wxIMAGE_OPTION_CUR_HOTSPOT_Y, hotSpotY ) ;
+        }
+        else
+        {
+            fileName.SetExt( "cur" );
+            image.LoadFile( fileName.GetFullPath(), wxBITMAP_TYPE_CUR );
+        }
+        if( image.IsOk() )
+        {
+            m_refData->DecRef();
+            m_refData = nullptr;
+            InitFromImage( image );
+        }
+        else
+            wxLogError( "Neither PNG nor CUR cursor image found in Resources" );
```

Comment:
Not sure if `wxLogError` should be used here. You might not want to annoy the user with error popups. I'd just use `wxFAIL_MSG` like it did before to inform the developer.

Reply:
@MaartenBent ,
Are you talking about end-user or the developr using the library?
How else the developer will know they need to add cursor resource to the Bundle?

OTOH, I agree it's annoying.

What's the good compromise here?

Reply:
The end-user shouldn't get automatically annoying popups if a cursor fails to load, so don't use `wxLogError`.
The developer will know it fails because `wxFAIL_MSG` will generate an assert or debug output or something. Like it was before with the `wxFAIL_MSG( wxT("Not implemented") );`

Reply:
@MaartenBent ,
Changed in the upcoming commit.